File: Ahorn/entities/SparkyDust.jl
module SparkyDust

using ..Ahorn, Maple

@mapdef Entity "MaggyHelper/SparkyDust" SparkyDust(x::Integer, y::Integer, Player::Bool=false, isActive::Bool=true, particleColor::String="ffff00", particleCount::Integer=10, radius::Number=32.0, soundEffect::String="event:/game/general/thing_booped", sparkFrequency::Number=2.0)

const placements = Ahorn.PlacementDict(
    "normal" => Ahorn.EntityPlacement(SparkyDust),
    "magical" => Ahorn.EntityPlacement(SparkyDust)
)

function Ahorn.selection(entity::SparkyDust)
    return Ahorn.Rectangle(entity.x - 8, entity.y - 8, 16, 16)
end

function Ahorn.render(ctx::Ahorn.Cairo.CairoContext, entity::SparkyDust, room::Maple.Room)
    Ahorn.drawSprite(ctx, "objects/IngesteHelper/sparky_dust", entity.x, entity.y)
end

end

File: framework/extracted/Loenn/entities/sparky_dust.lua
local sparkeyDust = {}

sparkeyDust.name = "MaggyHelper/SparkyDust"
sparkeyDust.depth = -100
sparkeyDust.texture = "objects/IngesteHelper/sparky_dust"
sparkeyDust.justification = {0.5, 0.5}

sparkeyDust.fieldInformation = {
    particleColor = {
        fieldType = "color"
    },
    particleCount = {
        fieldType = "integer",
        minimumValue = 1,
        maximumValue = 50
    },
    sparkFrequency = {
        fieldType = "number",
        minimumValue = 0.1,
        maximumValue = 10.0
    },
    radius = {
        fieldType = "number",
        minimumValue = 8.0,
        maximumValue = 100.0
    },
    followKglobal::global::Celeste.Player= {
        fieldType = "boolean"
    },
    isActive = {
        fieldType = "boolean"
    },
    soundEffect = {
        fieldType = "string",
        options = {
            "guid://{5a1e6a52-fa6a-44fb-b7ef-931a000b7c95}",
            "guid://{16b40879-0a79-4e42-8c91-fe419a8e186c}",
            "guid://{6160ad7b-16f6-49a5-aa9b-55d75da5a8e1}"
        },
        editable = true
    }
}

sparkeyDust.placements = {
    {
        name = "normal",
        data = {
            particleColor = "ffff00",
            particleCount = 10,
            sparkFrequency = 2.0,
            radius = 32.0,
            followKglobal::global::Celeste.Player= false,
            isActive = true,
            soundEffect = "guid://{5a1e6a52-fa6a-44fb-b7ef-931a000b7c95}"
        }
    },
    {
        name = "magical",
        data = {
            particleColor = "8844ff",
            particleCount = 20,
            sparkFrequency = 5.0,
            radius = 48.0,
            followKglobal::global::Celeste.Player= true,
            isActive = true,
            soundEffect = "guid://{6160ad7b-16f6-49a5-aa9b-55d75da5a8e1}"
        }
    }
}

return sparkeyDust
